src/api/procedures/redeemToken.ts

```diff
@@ -32,12 +32,16 @@ export async function prepareRedeemToken(
   const securityToken = new SecurityToken({ ticker }, context);
   const rawTicker = stringToTicker(ticker, context);
 
-  const { owner, primaryIssuanceAgent, isDivisible } = await securityToken.details();
+  const { primaryIssuanceAgents, isDivisible } = await securityToken.details();
 
-  const defaultPortfolio = new DefaultPortfolio(
-    { did: primaryIssuanceAgent ? primaryIssuanceAgent.did : owner.did },
-    context
-  );
+  if (primaryIssuanceAgents.length !== 1) {
+    throw new PolymeshError({
+      code: ErrorCode.ValidationError,
+      message: 'There is no a default Primary Issuance Agent for the given asset',
+    });
+  }
+
+  const defaultPortfolio = new DefaultPortfolio({ did: primaryIssuanceAgents[0].did }, context);
```

Comment:
Since `getAuthorizations` checks that the current identity is a PIA, we can just use the current DID here